conf/machine/include/imx-base.inc

```diff
@@ -403,6 +403,21 @@ PREFERRED_PROVIDER_virtual/kernel ??= "${IMX_DEFAULT_KERNEL}"
 SOC_DEFAULT_IMAGE_FSTYPES = "wic.bmap wic.gz"
 SOC_DEFAULT_IMAGE_FSTYPES_mxs = "uboot-mxsboot-sdcard wic.bmap wic.gz"
 
+# Add support for Syslinux to mainline BSP.
+# U-Boot has the Distro Boot mode enabled by default, which
+# require that either Syslinux to be enabled, or a boot script
+# to be used to define the boot process.
+# We opt-in for Syslinux, since it is designated as a preferred
+# distro boot mode according to the U-Boot documentation.
+UBOOT_EXTLINUX_use-mainline-bsp ?= "1"
+UBOOT_EXTLINUX_LABELS_use-mainline-bsp ?= "default"
+UBOOT_EXTLINUX_DEFAULT_LABEL_use-mainline-bsp ?= "i.MX8M Family EVK"
+
+UBOOT_EXTLINUX_FDT_default_use-mainline-bsp     ?= "../${KERNEL_DEVICETREE_BASENAME}.dtb"
+UBOOT_EXTLINUX_CONSOLE_default_use-mainline-bsp ?= "console=${console}"
+UBOOT_EXTLINUX_ROOT_default_use-mainline-bsp    ?= "root=/dev/mmcblk1p2"
+UBOOT_EXTLINUX_MENU_DESCRIPTION_default_use-mainline-bsp ?= "i.MX8M Family EVK"
```

Comment:
I'd prefer this to be added to every machine as it is hard to come with a generic solution here.

Reply:
Sure, no problem! Funny enough - I had it originally duplicated in every machine include file, but decided to unify parts that were identical across. 😄 

I would move it back to respective machine include files and update this PR with _V2_.